jwe/jwe.go

```diff
@@ -171,40 +171,119 @@ func Encrypt(payload []byte, keyalg jwa.KeyEncryptionAlgorithm, key interface{},
 	return Compact(msg)
 }
 
+// DecryptCtx is used internally when jwe.Decrypt is called. Regular users
+// should not have to touch this object, but if you need advanced handling
+// of messages, you might have to use it. Only use it when you really
+// understand how JWE processing works in this library.
+type DecryptCtx interface {
+	Algorithm() jwa.KeyEncryptionAlgorithm
+	SetAlgorithm(jwa.KeyEncryptionAlgorithm)
+	Key() interface{}
+	SetKey(interface{})
+	Message() *Message
+	SetMessage(*Message)
+}
+
+type decryptCtx struct {
+	alg jwa.KeyEncryptionAlgorithm
+	key interface{}
+	msg *Message
+}
+
+func (ctx *decryptCtx) Algorithm() jwa.KeyEncryptionAlgorithm {
+	return ctx.alg
+}
+
+func (ctx *decryptCtx) SetAlgorithm(v jwa.KeyEncryptionAlgorithm) {
+	ctx.alg = v
+}
+
+func (ctx *decryptCtx) Key() interface{} {
+	return ctx.key
+}
+
+func (ctx *decryptCtx) SetKey(v interface{}) {
+	ctx.key = v
+}
+
+func (ctx *decryptCtx) Message() *Message {
+	return ctx.msg
+}
+
+func (ctx *decryptCtx) SetMessage(m *Message) {
+	ctx.msg = m
+}
+
+// PostParser is called right after the JWE message has been parsed but
+// before the actual decryption takes place during `jwe.Decrypt()`.
+type PostParser interface {
+	Do(DecryptCtx) error
```

Comment:
Could you please add a few examples/tests for this interface?

Is it something I should use for clevis.go here https://github.com/anatol/clevis.go/blob/main/tang.go#L56 ?

Reply:
Oh oops, I forgot to add anything there didn't I. And I realize the interface is incomplete. Fix coming up...

Reply:
@anatol hopefully this will get you going https://github.com/lestrrat-go/jwx/pull/406/commits/c357f46ed320928737f10015aaac6efdd827137a